client/settings/payment-settings/promotional-banner/re-connect-account-banner.js

```diff
@@ -14,32 +15,57 @@ import {
 	NewPill,
 } from 'wcstripe/settings/payment-settings/promotional-banner/banner-layout';
 
-export const ReConnectAccountBanner = ( { testOauthUrl, oauthUrl } ) => {
+export const ReConnectAccountBanner = () => {
 	const [ isTestModeEnabled ] = useTestMode();
 	const { createErrorNotice } = useDispatch( 'core/notices' );
+	const [ isLoading, setIsLoading ] = useState( false );
 
-	const handleButtonClick = () => {
-		if ( isTestModeEnabled && testOauthUrl ) {
-			recordEvent( 'wcstripe_create_or_connect_test_account_click', {} );
-			recordEvent( 'wcstripe_reconnect_button_click', {
-				source: 're-connect-account-banner',
-				mode: 'test',
-			} );
-			window.location.assign( testOauthUrl );
-		} else if ( ! isTestModeEnabled && oauthUrl ) {
-			recordEvent( 'wcstripe_create_or_connect_account_click', {} );
-			recordEvent( 'wcstripe_reconnect_button_click', {
-				source: 're-connect-account-banner',
-				mode: 'live',
+	const handleButtonClick = async () => {
+		const mode = isTestModeEnabled ? 'test' : 'live';
+
+		recordEvent(
+			isTestModeEnabled
+				? 'wcstripe_create_or_connect_test_account_click'
+				: 'wcstripe_create_or_connect_account_click',
+			{}
+		);
+		recordEvent( 'wcstripe_reconnect_button_click', {
+			source: 're-connect-account-banner',
+			mode,
+		} );
+
+		setIsLoading( true );
+
+		try {
+			const response = await jQuery.ajax( {
+				url: ajaxurl,
+				method: 'POST',
+				data: {
+					action: 'wc_stripe_get_oauth_url',
+					mode,
+					nonce: wc_stripe_settings_params.oauth_nonce, // eslint-disable-line camelcase
+				},
 			} );
-			window.location.assign( oauthUrl );
-		} else {
+
```

Comment:
Minor nit: is it worth setting the loading state to false here regardless of the result?
```suggestion

			setIsLoading( false );
```

Reply:
We want to keep the button `disbaled` while we redirect (`isLoading === true`)... to avoid the possibility of additional clicks triggering another request and generating multiple URLs (and triggering the state invalidation bug)